gdal/ogr/ogr_srs_esri.cpp

```diff
@@ -1835,7 +1841,10 @@ OGRErr OGRSpatialReference::morphToESRI()
       }
       if( pszGcsName != nullptr )
       {
-        RemapGeogCSName(this, pszGcsName);
+        if( RemapGeogCSName(this, pszGcsName) < 0 )
```

Comment:
I think this test should be removed. I'm not even sure the return code from RemapGeogCSName() is meaningful at all

Reply:
That indeed fixes the test failure.  RemapGeogCSName is definitely confusing.